UITests/BookmarksPanelTests.swift

```diff
@@ -83,26 +103,45 @@ class BookmarksPanelTests: KIFTestCase {
             return
         }
         
-        let changedBufferRecords = [
-            BookmarkMirrorItem.bookmark("guid0", modified: Date.now(), hasDupe: false, parentID: BookmarkRoots.MobileFolderGUID, parentName: nil, title: "xyz", description: nil, URI: "http://unused.com", tags: "[]", keyword: nil),
-            BookmarkMirrorItem.folder(BookmarkRoots.MobileFolderGUID, modified: Date.now(), hasDupe: false, parentID: BookmarkRoots.RootGUID, parentName: nil, title: "", description: nil, children: ["guid0"])
-        ]
-        XCTAssert(bookmarks.applyRecords(changedBufferRecords).value.isSuccess)
-        
+        // TEST: Create remote mobile bookmark, and verify the bookmark appears in the root and the remote mobile folder is not shown
+        var applyResult = bookmarks.applyRecords([
+            makeBookmark(guid: "bm-guid0", parentID: BookmarkRoots.MobileFolderGUID, title: "xyz"),
+            makeFolder(guid: BookmarkRoots.MobileFolderGUID, parentID: BookmarkRoots.RootGUID, title: "", childrenGuids: ["bm-guid0"])
+            ])
+        XCTAssert(applyResult.value.isSuccess)
         EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("Bookmarks")).perform(grey_tap())
         
         // is this in the root?
-        EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("xyz")).assert(grey_notNil())
+        assertRowExists(withTitle: "xyz")
         
-        EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("Desktop Bookmarks")).perform(grey_tap())
+        navigateFolder(withTitle: "Desktop Bookmarks")
         // this should be missing, they are shown in the root
         EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("Mobile Bookmarks")).assert(grey_nil())
         
-        // Add a local bookmark, and then navigate back to the root view (the navigation will refresh the table).
-        let ok = (bookmarks as! MergedSQLiteBookmarks).local.addToMobileBookmarks(URL(string: "http://another-unused")!, title: "123", favicon: nil)
-        XCTAssert(ok.value.isSuccess)
-        EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("Bookmarks")).inRoot(grey_kindOfClass(NSClassFromString("UITableView")!)).perform(grey_tap())
+        // TEST: Add a local bookmark, and then navigate back to the root view (the navigation will refresh the table).
+        let isAdded = (bookmarks as! MergedSQLiteBookmarks).local.addToMobileBookmarks(URL(string: "http://another-unused")!, title: "123", favicon: nil)
+        XCTAssert(isAdded.value.isSuccess)
+        navigateBackInTableView()
         // is this in the root?
-        EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("123")).assert(grey_notNil())
+        assertRowExists(withTitle: "123")
+        
+        // TEST: Add sub-folder to MobileFolderGUID, ensure it is navigable
```

Comment:
I should have split up the PR, this part is the only new code, the rest is just refactoring this test code for reuse.